tests/test_cli.py

```diff
@@ -0,0 +1,131 @@
+        engine = sa.create_engine(postgresql.url())
+
+        create_table_process = await asyncio.create_subprocess_exec(
+            "explog",
+            "create-table",
+            stderr=subprocess.PIPE,
+        )
+
+        await asyncio.wait_for(
+            create_table_process.wait(), timeout=CREATE_TIMEOUT
+        )
+        data = await create_table_process.stderr.read()  # type: ignore
+        assert (
+            create_table_process.returncode == 0
+        ), f"explog create-table failed with {data.decode()!r}"
+
+        table_names = sa.inspect(engine).get_table_names()
+        assert "exposure_log_messages" in table_names
+
+        for port, message_id in ((None, 1), (8000, 2)):
+            await check_run_and_add_message(port=port, message_id=message_id)
+
+
+async def check_run_and_add_message(
+    port: Optional[int], message_id: int
+) -> None:
+    """Run `explog run` and use it to add one message.
+
+    Before calling this you must have a database running.
+
+    Parameters
+    ----------
+    port
+        Port on which to run the explog service.
+    message_id
+        Expected ID of the added message.
+    """
+
+    cmdline_args = ["explog", "run"]
+    if port is None:
+        port = 8080
+    else:
+        cmdline_args += ["--port", str(port)]
+
+    run_process = await asyncio.create_subprocess_exec(
+        *cmdline_args,
+        stderr=subprocess.PIPE,
+    )
+    try:
+
+        # Give the exposure log service time to start.
+        await asyncio.sleep(RUN_DELAY)
+
+        # Add a message whose obs_id matches an exposure.
+        add_args = dict(
+            obs_id="HSCA90333600",
+            instrument="HSC",
+            message_text="A sample message",
+            user_id="test_add_message",
+            user_agent="pytest",
+            is_human=False,
+            is_new=False,
+            exposure_flag="none",
+        )
+        add_data, headers = format_http_request(
+            category="mutation",
+            command="add_message",
+            args_dict=add_args,
+            fields=["id"],
+        )
+        r = requests.post(
+            f"http://localhost:{port}/explog/graphql",
+            add_data,
+            headers=headers,
+        )
+        assert r.status_code == 200
+        reply_data = r.json()
+        assert "error" not in reply_data
+        assert reply_data["data"]["add_message"]["id"] == message_id
+    finally:
+        if run_process.returncode is None:
+            run_process.terminate()
+            await asyncio.wait_for(
+                run_process.wait(), timeout=TERMINATE_TIMEOUT
+            )
+        else:
+            try:
+                stderr_bytes = await run_process.stderr.read()  # type: ignore
+                stderr_msg = stderr_bytes.decode()
+            except Exception as e:
+                stderr_msg = f"could not read stderr: {e}"
+            raise AssertionError(f"run_process failed: {stderr_msg}")
```

Comment:
It seems to me like you would only want to raise if you get an `Exception`.  This appears to raise any time the code gets to the `else` block.

Reply:
This test expects that the `explog run` process is still be running at this point in the code. That's the if condition.
If the process has exited (the `else` condition) then the test fails. I make a "best effort" attempt to stderr from the process and include it in the error message, but I am going to report the failure with or without that information.

I'll add a code comment explaining.

Reply:
Thank you.  Sounds good.